erpnext/selling/doctype/quotation/quotation.json

```diff
@@ -2867,6 +2897,66 @@
    "search_index": 0, 
    "set_only_once": 0, 
    "unique": 0
+  }, 
+  {
+   "allow_bulk_edit": 0, 
+   "allow_on_submit": 0, 
+   "bold": 0, 
+   "collapsible": 0, 
+   "columns": 0, 
+   "fieldname": "applied_coupon_code", 
+   "fieldtype": "Data", 
+   "hidden": 0, 
+   "ignore_user_permissions": 0, 
+   "ignore_xss_filter": 0, 
+   "in_filter": 0, 
+   "in_global_search": 0, 
+   "in_list_view": 0, 
+   "in_standard_filter": 0, 
+   "label": "Applied Coupon Code", 
+   "length": 0, 
+   "no_copy": 0, 
+   "permlevel": 0, 
+   "precision": "", 
+   "print_hide": 0, 
+   "print_hide_if_no_value": 0, 
+   "read_only": 0, 
+   "remember_last_selected_value": 0, 
+   "report_hide": 0, 
+   "reqd": 0, 
+   "search_index": 0, 
+   "set_only_once": 0, 
+   "unique": 0
+  }, 
+  {
+   "allow_bulk_edit": 0, 
+   "allow_on_submit": 0, 
+   "bold": 0, 
+   "collapsible": 0, 
+   "columns": 0, 
+   "fieldname": "total_amount_without_discount", 
```

Comment:
Why two extra field `total_amount_without_discount` and `discount`? These values can be calculated from the item tables in the client side itself (if your purpose is to show the discount amount in the shopping cart)

Reply:
Hello @manassolanki 
we take both field "total_amount_without_discount" and "discount" to keep the history and also purpose to show the discount amount in the shopping cart. it will become helpful for further and future enhancement. 

Quotation and Sales Order both views are separate and when the shopping cart page is rendered that type it will get the values from the doc (database) and display on screen through the .html template.

Thanks